test/actions/submit-drake.js

```diff
@@ -4,30 +4,89 @@ import * as actions from '../../src/code/actions';
+        wings: "Wings"
+      },
+      userAlleles = "a:T,b:T,a:M,b:M,a:W,b:W,a:h,b:h,a:C,b:C,a:B,b:B,a:Fl,b:Fl,a:hl,b:hl,a:A1,b:A1,a:D,a:Bog,a:rh",
+      initialAlleles = "a:T,b:T,a:M,b:M,a:W,b:w,a:h,b:h,a:C,b:C,a:B,b:B,a:Fl,b:Fl,a:hl,b:hl,a:A1,b:A1,a:D,a:Bog,a:rh",
+      getState = () => ({
+        routeSpec: {level: 0,mission: 0, challenge: 0}, 
+        drakes: [
+          {
+            phenotype: {characteristics: correctCharacteristics},
+            alleleString: correctAlleles,
+            sex: 0
+          },
+          {
+            phenotype: {characteristics: userCharacteristics},
+            alleleString: userAlleles,
+            sex: 0
+          }
+        ],
+        initialDrakes: [
+          {},
+          {
+            alleleString: initialAlleles
+          }
+        ],
+        trial: 0, trials: [{}], 
+        authoring: [[[{userChangeableGenes: "changeableGenes"}]]]
+      });
+
+
 describe('submitDrake action', () => {
   describe('when the drake is submitted correctly', () => {
-    const correctPhenotype = [{armor: "Five armor"}],
-          submittedPhenotype = [{armor: "Five armor"}],
-          correct = true;
-
-    const dispatch = expect.createSpy();
-    const getState = () => ({routeSpec: {level: 0,mission: 0, challenge: 0}, trial: 0, trials: [{}], authoring: [[{}]]});
+    const correct = true,
+          dispatch = expect.createSpy();
 
-    actions.submitDrake(correctPhenotype, submittedPhenotype, correct)(dispatch, getState);
+    actions.submitDrake(0, 1, correct)(dispatch, getState);
 
     it('should call dispatch with the correct _submitDrake action', () => {
-      expect(dispatch).toHaveBeenCalledWith({
-        type: types.DRAKE_SUBMITTED,
-        correctPhenotype,
-        submittedPhenotype,
-        correct,
-        incrementMoves: false,
-        meta: {
-          "itsLog": {
-            "actor": "USER",
-            "action": "SUBMITTED",
-            "target": "DRAKE"
-          }
+      var submitDrakeAction = dispatch.calls[0].arguments[0];
+      expect(submitDrakeAction.type).toEqual(types.DRAKE_SUBMITTED);
+      assertMatchingPhenotype(submitDrakeAction.correctPhenotype, correctCharacteristics);
+      assertMatchingPhenotype(submitDrakeAction.submittedPhenotype, userCharacteristics);
+      expect(submitDrakeAction.initialAlleles).toEqual(initialAlleles);
+      expect(submitDrakeAction.selectedAlleles).toEqual(userAlleles);
+      expect(submitDrakeAction.targetAlleles).toEqual(correctAlleles);
+      expect(submitDrakeAction.targetSex).toEqual(0);
+      expect(submitDrakeAction.submittedSex).toEqual(0);
+      expect(submitDrakeAction.editableGenes).toEqual("changeableGenes");
+      expect(submitDrakeAction.correct).toEqual(true);
+      expect(submitDrakeAction.incrementMoves).toEqual(false);
+      expect(submitDrakeAction.meta).toEqual({
+        "itsLog": {
+          "actor": "USER",
+          "action": "SUBMITTED",
+          "target": "DRAKE"
```

Comment:
I wonder if this long list of expects (here and for the incorrect action) would be more legible if the objects were compared directly, and doing the phenotype comparison separately. We've done that elsewhere, I don't know if you think it's more legible or more confusing.

The following passes:
```
      expect(submitDrakeAction).toEqual({
        type: types.DRAKE_SUBMITTED,
        species: "Drake",
        correctPhenotype: submitDrakeAction.correctPhenotype,  // we check valid phenotype below
        submittedPhenotype: submitDrakeAction.submittedPhenotype,
        initialAlleles: initialAlleles,
        selectedAlleles: userAlleles,
        targetAlleles: correctAlleles,
        targetSex: 0,
        submittedSex: 0,
        editableGenes: "changeableGenes",
        correct: true,
        incrementMoves: false,
        meta: {
          "itsLog": {
            "actor": "USER",
            "action": "SUBMITTED",
            "target": "DRAKE"
          }
        }
      });
      assertMatchingPhenotype(submitDrakeAction.correctPhenotype, correctCharacteristics);
      assertMatchingPhenotype(submitDrakeAction.submittedPhenotype, userCharacteristics);
      
```